ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveRexExecutorImpl.java

```diff
@@ -52,6 +56,10 @@ public void reduce(RexBuilder rexBuilder, List<RexNode> constExps, List<RexNode>
       // initialize the converter
       ExprNodeConverter converter = new ExprNodeConverter("", null, null, null,
           new HashSet<>(), rexBuilder.getTypeFactory());
+
+      if (rexNode.getKind() == SqlKind.IS_DISTINCT_FROM) {
```

Comment:
This should be move to the `ExprNodeConverter` itself. For instance, check `visitCall` method.

Reply:
Moved to `ExprNodeConverter`; passed `rexBuilder` through a new constructor.